pkg/tenant/tenant.go

```diff
@@ -0,0 +1,86 @@
+package tenant
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+)
+
+var (
+	errTenantIDTooLong = errors.New("tenant ID is too long: max 150 characters")
+)
+
+type errTenantIDUnsupportedCharacter struct {
+	pos      int
+	tenantID string
+}
+
+func (e *errTenantIDUnsupportedCharacter) Error() string {
+	return fmt.Sprintf(
+		"tenant ID '%s' contains unsupported character '%c'",
+		e.tenantID,
+		e.tenantID[e.pos],
+	)
+}
+
+const tenantIDsLabelSeparator = "|"
+
+// NormalizeTenantIDs is putting the IDs in a normalized form, which is sorts and de-duplicate them
+func NormalizeTenantIDs(ids []string) []string {
+	sort.Strings(ids)
+
+	// de-duplicate orgIDs
+	var result []string
```

Comment:
Performance wise a couple of things:
1. The input `ids` is just fine if there are no duplicates (which I would expect to be the common case)
2. `result` make be pre-allocated with the length of `ids`